enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,517 @@
+
+## Proposal
+
+- Create a new `DefaultPlacement` API field.
+
+- Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
+
+### API Extensions
+
+Introduce a new status field in the Ingress config CR
+(`config.openshift.io/v1/ingresses`) called `DefaultPlacement`.
+
+In addition, continue to allow the `.spec.replicas` and `.spec.nodePlacement`
+fields in `operator.openshift.io/v1/ingresscontrollers` CRs to be omitted, but
+change the defaulting behavior for these fields based on the new status field.
+
+The sections below go into detail about the field, its possible values, and the
+behavior expected from each of them, but in practice following is the proposed
+addition to `openshift/api`'s `config/v1/types_ingress.go` file:
+
+```go
+type IngressStatus struct {
+    // ... existing fields omitted
+
+	// defaultPlacement is set by the installer and lets it control which
+	// nodes will host the ingress router pods by default, the options being
+	// control-plane nodes or worker nodes.
+	//
+	// This field works by dictating how the Ingress Operator will set the default
+	// values of future IngressController resources' replicas and nodePlacement
+	// fields.
+	//
+	// The value of replicas is set based on the value of a chosen field in the
+	// Infrastructure CR. If defaultPlacement was set to ControlPlane, the
+	// chosen field will be controlPlaneTopology. If it was set to Workers the
+	// chosen field will be infrastructureTopology. Replicas will then be set to 1
+	// or 2 with respect to the chosen field's value (which can be either
+	// SingleReplica or HighlyAvailable).
+	//
+	// The value of nodePlacement is adjusted based on defaultPlacement - If
+	// defaultPlacement is set to ControlPlane the "node-role.kubernetes.io/worker"
+	// label will be added. If defaultPlacement is set to Workers the
+	// "node-role.kubernetes.io/master" label will be added.
+	//
+	// +kubebuilder:validation:Enum:="ControlPlane";"Workers"
+	// +kubebuilder:default:="Workers"
+	// +optional
+	DefaultPlacement DefaultPlacement `json:"defaultPlacement"`
+}
+
+// DefaultPlacement defines the default placement of ingress router pods.
+type DefaultPlacement string
+
+const (
+	// "Workers" is for having router pods placed on worker nodes by default
+	Workers DefaultPlacement = "Workers"
+
+	// "ControlPlane" is for having router pods placed on control-plane nodes by default
+	ControlPlane DefaultPlacement = "ControlPlane"
+)
```

Comment:
Very much a nit, just thinking about how these will be imported, `ingressv1.Workers`, `ingressv1.ControlPlane`, wondering if you might want to make the names more descriptive, eg, `type DefaultPlacementTopology string` and `WorkersPlacementTopology`/`ControlPlanePlacmentTopology`. I don't have a strong opinion here though and this can be hashed out during the API PR